src/zkevm_specs/state.py

```diff
@@ -182,6 +239,12 @@ def check_storage(row: Row, row_prev: Row):
     if not all_keys_eq(row, row_prev):
         assert row.is_write == 1 and row.rw_counter == 0
 
+    # 2. MPT storage lookup with incremental counter
+    value_prev = row_prev.value if all_keys_eq(row, row_prev) else FQ(0)
```

Comment:
After thinking about this and following our chat discussion I've made some changes:
1. I noticed that we had a single column for the tx_id in the storage, but we can just use the id column (key1), so I moved it there
2. We already have a `committed_value` column, which must match the previous value before the tx.  If we add tx to the keys, this means that on the first entry for a set of keys, the value_prev that must be used in the MPT lookup is the `committed_value`, so that's the value used in the lookup now in such chase.  This allows us to remove the need to add artificial rows for the first access, because we can already construct such lookup no matter if it's READ or WRITE by using the `commited_value` as `value_prev`.  At the same time, this makes the rules for `rw_counter` simpler!  Previously we were using `rw_counter = 0` for such artificial rows, but there was no constraint forbidding the usage of `rw_counter = 0` elsewhere!  So a malicious prover could insert operations freely.  Now `rw_counter` must be != 0 except for `Tag.Start`.